build: use Ninja on travis CI; avoid parallel Luarocks bootstrap

justinmk opened the pull request:
ref https://github.com/neovim/neovim/pull/8084

justinmk committed:
test: handle non-deterministic message cadence

justinmk committed:
ci/AppVeyor: fix `set` whitespace quoting

justinmk committed:
ci/AppVeyor: disable MSVC_32 build

The MSVC_32 currently hangs.  When MSVC becomes the primary Windows
target, we can enable MSVC_32 and retire one of the mingw builds.  In
the meantime it adds too much time.

b-r-o-c-k commented on ci/build.bat:
```diff
@@ -6,19 +6,19 @@ if "%CONFIGURATION%" == "MINGW_32" (
   :: MSYS2, this allows using all the dependencies and tools available
   :: in MSYS2, but we cannot build inside the MSYS2 shell.
-  set CMAKE_GENERATOR="MinGW Makefiles"
+  set "CMAKE_GENERATOR=MinGW Makefiles"
```
The quotes need to be part of the variable's value. They can be escaped like this:
```batch
set "CMAKE_GENERATOR=^"MinGW Makefiles^""
```
Or you could put quotes around the variable reference like this:
```batch
cmake -G "%CMAKE_GENERATOR%"
```

Would you accept a PR replacing this Batch script with a PowerShell script? I think it would be much easier to work with.

justinmk replied:
> Would you accept a PR replacing this Batch script with a PowerShell script?

Definitely!

janlazo replied:
The carets are unnecessary because it's the last argument. When using carets, it's best to use it for each metacharacter, including the double quotes used for wrapping the argument, in the entire command.

`set "CMAKE_GENERATOR="MinGW Makefiles""` is fine.

justinmk committed:
build/luarocks: apply "Fix siteconfig" patch

upstream: https://github.com/luarocks/luarocks/pull/774

justinmk committed:
test: rename next_message() to next_msg()

justinmk committed:
test: next_msg(): default `timeout` to 10s

Infinite timeout results in hangs which waste time. If some test needs
longer than 10s to wait for a message, it should specify the timeout
explicitly.

justinmk committed:
build/luarocks: avoid parallelism for luarocks build

Is there a race between the luarocks `make bootstrap` dependencies?

reverts f73b4911312b35bfe38ed068672a2f8ba8875ba7
ref https://github.com/luarocks/luarocks/pull/774

justinmk committed:
ci/travis: use ninja instead of make

janlazo commented:
Is MSVC build always hanging on `test/functional\terminal\ex_terminal_spec.lua`?

janlazo commented:
Why not disable `MSVC_32` since `MSVC_64` doesn't hang?

justinmk committed:
ci/macOS: skip python2 on travis macOS

macOS travis builds recently started failing (travis caches were cleared
recently, maybe related). python2 is reasonably covered by linux CI. Not
going to waste time on it for macOS CI.

    ==> Installing python@2
    ==> Downloading https://homebrew.bintray.com/bottles/python@2-2.7.14_3.el_capita
    ==> Pouring python@2-2.7.14_3.el_capitan.bottle.tar.gz
    Error: The `brew link` step did not complete successfully
    The formula built, but is not symlinked into /usr/local
    Could not symlink bin/2to3-2
    Target /usr/local/bin/2to3-2
    is a symlink belonging to python. You can unlink it:
      brew unlink python
    To force the link and overwrite all conflicting files:
      brew link --overwrite python@2
    To list all files that would be deleted:
      brew link --overwrite --dry-run python@2
    Possible conflicting files are:
    /usr/local/bin/2to3-2 -> /usr/local/Cellar/python/2.7.12_1/bin/2to3-2
    /usr/local/bin/2to3-2.7 -> /usr/local/Cellar/python/2.7.12_1/bin/2to3-2.7
    /usr/local/bin/idle -> /usr/local/Cellar/python/2.7.12_1/bin/idle
    ...

justinmk committed:
build: respect $DEPS_BUILD_DIR

Without this, the CI_TARGET=lint travis job cant't find the cached deps
(in $HOME/nvim-deps), nor can it update the cache.

justinmk committed:
ci/travis: rename $BUILD_NVIM_DEPS to $CACHE_ENABLE
